instrumentation/spring/spring-cloud-gateway/spring-cloud-gateway-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/v2_0/ServerWebExchangeHelper.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.v2_0;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.web.server.ServerWebExchange;
+
+public final class ServerWebExchangeHelper {
+
+  /** Route info key. */
+  public static final String ROUTE_INFO_ATTRIBUTES = "ROUTE_INFO";
+
+  private ServerWebExchangeHelper() {}
+
+  public static void extractAttributes(ServerWebExchange exchange, Context context) {
+    // Record route info
+    Route route = exchange.getAttribute(GATEWAY_ROUTE_ATTR);
+    if (route != null) {
+      Span currentSpan = Span.fromContext(context);
+      assert currentSpan != null;
+      currentSpan.setAttribute(ROUTE_INFO_ATTRIBUTES, route.toString());
+    }
+  }
+
+  public static String extractServerRoute(ServerWebExchange exchange) {
+    Route route = exchange.getAttribute(GATEWAY_ROUTE_ATTR);
+    if (route != null) {
+      return "Route@" + route.getId();
```

Comment:
The problem with route id is that it may have high cardinality, if not specified by user it defaults to an UUID. Spec requires `http.route` to have a low cardinality.

Reply:
I actually don't think using `routeId` will result in excessive cardinality. Users typically redirect multiple Http Paths with a single route. So I think the cardinality of `routeId` is lower than the cardinality of `HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE`  in spring-cloud-gateway senario.

Reply:
See https://github.com/spring-cloud/spring-cloud-gateway/blob/51367a8668c686e972f621874ec8e40ae21b56b2/spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/route/builder/RouteLocatorBuilder.java#L82 As far as I can tell if the user does not specify the route id a random one is generated, you need to handle that. If the route id is random you need to ignore it.

Reply:
> See https://github.com/spring-cloud/spring-cloud-gateway/blob/51367a8668c686e972f621874ec8e40ae21b56b2/spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/route/builder/RouteLocatorBuilder.java#L82 As far as I can tell if the user does not specify the route id a random one is generated, you need to handle that. If the route id is random you need to ignore it.

I get it！The spring-cloud-gateway may generate random route id everytime the route is loaded if the route id is not set. I will handle that.